ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -925,27 +929,25 @@ private CastVisitor() {
     }
 
     static Pair<String, SqlType> getCast(final Pair<String, SqlType> expr, final SqlType sqlType) {
-      if (!sqlType.supportsCast()) {
-        throw new KsqlFunctionException(
-            "Only casts to primitive types and decimal are supported: " + sqlType);
-      }
-
-      final SqlType rightSchema = expr.getRight();
-      if (sqlType.equals(rightSchema) || rightSchema == null) {
+      final SqlType sourceType = expr.getRight();
+      if (sourceType == null || sqlType.equals(sourceType)) {
+        // sourceType is null if source is SQL NULL
         return new Pair<>(expr.getLeft(), sqlType);
       }
 
-      return CASTERS.getOrDefault(
-          sqlType.baseType(),
-          (e, t, r) -> {
-            throw new KsqlException("Invalid cast operation: " + t);
-          }
-      )
-          .cast(expr, sqlType, sqlType);
+      return CASTERS.getOrDefault(sqlType.baseType(), CastVisitor::unsupportedCast)
+          .cast(expr, sqlType);
+    }
```

Comment:
getCast no longer calls sqlType.supportsCast as it is this the getCast method that determines which casts are supported and which are not. Hence supportsCast is superfluous and likely to get out of date with what this method supports.